tool/generator/formatting.dart

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+List<String> formatDocs(String data, int width) {
+  final lines = data.split('\n');
+  final output = <String>[];
+
+  width -= '/// '.length;
+
+  var inCodeFence = false;
+
+  for (var line in lines) {
+    if (inCodeFence) {
+      output.add(line);
+      if (line.startsWith('```')) {
+        inCodeFence = false;
+      }
+    } else {
+      final ltrim = line.trimLeft();
+
+      if (line.startsWith('> ')) {
+        final wrapped = _wrap(line.substring(2), width - 2);
+        output.addAll(wrapped.map((l) => '> $l'));
+      } else if (line.startsWith('- ')) {
+        final wrapped = _wrap(line.substring(2), width - 2);
+        output.add('- ${wrapped.first}');
+        output.addAll(wrapped.skip(1).map((l) => '  $l'));
+      } else if (ltrim.startsWith('- ') || ltrim.startsWith('<')) {
+        output.add(line);
+      } else {
+        output.addAll(_wrap(line, width));
+
+        if (line.startsWith('```')) {
+          inCodeFence = true;
+        }
+      }
+    }
+  }
+
+  return output.map((line) => line.isEmpty ? '///' : '/// $line').toList();
+}
+
+final RegExp _linksRegEx = RegExp(r'\[([\w `-]+?)\](\(\S+\))?');
+
+Iterable<String> _wrap(String line, int width) sync* {
```

Comment:
Hah, I implemented something similar in translator.dart for the HTML constructors here: https://github.com/dart-lang/web/pull/185. I'll see if I can reuse this code instead. Can you add some doc comments here and above?